manifests/install/windows.pp

```diff
@@ -81,7 +106,9 @@
     path    => $::path,
     require => [
       Puppet_agent_upgrade_error['puppet_agent_upgrade_failure.log'],
-      File[$_installps1]
+      File[$_installps1],
+      File[$_prerequisites_check],
+      Exec['prerequisites_check.ps1']
```

Comment:
Isn't it enough to specify the Exec resource here? Since the Exec itself depends on `File[$_prerequisites_check]`. Just a thought.